samples/hello_world/sample.yaml

```diff
@@ -6,11 +6,15 @@ common:
   tags: introduction
   integration_platforms:
     - native_posix
-  harness: console
-  harness_config:
-    type: one_line
-    regex:
-      - "Hello World! (.*)"
 tests:
   sample.basic.helloworld:
     tags: introduction
+    harness: console
+    harness_config:
+      type: one_line
+      regex:
+        - "Hello World! (.*)"
+  sample.basic.helloworld.robot:
```

Comment:
@hakehuang we were thinking about this and we believe there are a few arguments against this "implicit" approach:

1. This will be a significant change to the way we generate a list of tests in Twister - we definitely would like to be able to run all test cases that are available, including both "classic" tests and Robot tests
2. With the new support we will be able to run tests that are not really similar to the ones we already have. It will be difficult to decide what to do with harness parameters, for example.
3. It will be quite common to have multiple Robot files per application - possibly with some parameters required etc. I think it will be actually quite **rare** to have a situation like this, where the Robot file would resemble the current test 1:1. We treat the hello-world more as a usage example, rather than added value. If you'd be more comfortable with removing the hello-world test case completely, we can of course do that.

In general I believe this automated approach you propose is worth having a further discussion, but I am not sure if this should be the first step?